src/modules/rover_pos_control/RoverPositionControl.cpp

```diff
@@ -455,13 +455,20 @@ RoverPositionControl::run()
 			//orb_copy(ORB_ID(vehicle_attitude), _vehicle_attitude_sub, &_vehicle_att);
 			_act_controls.timestamp = hrt_absolute_time();
 
-			if (_actuator_controls_pub != nullptr) {
-				//PX4_INFO("Publishing actuator from pos control");
-				orb_publish(ORB_ID(actuator_controls_0), _actuator_controls_pub, &_act_controls);
+			/* Only publish if any of the proper modes are enabled */
+			if (_control_mode.flag_control_velocity_enabled ||
+			    _control_mode.flag_control_rates_enabled ||
+			    _control_mode.flag_control_attitude_enabled ||
```

Comment:
Have you tested this in mission mode? I think this will prevent you from using auto mode

Reply:
We also don't want   `flag_control_rates_enabled` and `flag_control_attitude_enabled` since they are not supported

Reply:
I created a quick mission in QGC and the rover did it well (using gazebo SITL). So this should be fine? Or what is auto mode?
I just deleted the two flags in the next commit.

Reply:
I was assuming https://github.com/PX4/Firmware/blob/07d656e971a72d1202651dfd3b4642736fb078d7/msg/vehicle_control_mode.msg#L7 will be enabled for mission mode. How about offboard position setpoints?

Reply:
As far as I see, the Commander always sets `flag_control_auto_enabled` together with `flag_control_rates_enabled` `flag_control_attitude_enabled` and `flag_control_climb_rate_enabled` for the most of the AUTO_* navigation states, which includes `NAVIGATION_STATE_AUTO_MISSION`.
But for missions, `flag_control_position_enabled` and `flag_control_velocity_enabled` are also set to true, if you're not in transition mode (for VTOL I guess).

Offboard position setpoints are also working fine. But why is the rover loitering at each position waypoint and also when using Land mode or RTL? But I think, that should be another issue.